File: scrumly-backend/scrumly-api/conference-service/src/main/java/com/scrumly/conferenceservice/services/ConferenceService.java
package com.scrumly.conferenceservice.services;

import com.scrumly.conferenceservice.dto.ConferenceConfigDto;
import com.scrumly.conferenceservice.dto.ConferenceRoomDto;
import com.scrumly.conferenceservice.dto.user.UserProfileDto;
import com.scrumly.conferenceservice.utils.RetryException;
import io.openvidu.java.client.OpenViduHttpException;
import io.openvidu.java.client.OpenViduJavaClientException;

import java.util.List;

public interface ConferenceService {
    String createConference(String workspaceId);

    void deleteConference(String workspaceId);

    ConferenceRoomDto joinConference(String conferenceId) throws OpenViduJavaClientException, OpenViduHttpException, RetryException, InterruptedException;

    void exitConference(String conferenceId) throws OpenViduJavaClientException, OpenViduHttpException;

    void kickFromConference(String conferenceId, String userId) throws OpenViduJavaClientException, OpenViduHttpException;

    void closeConference(String conferenceId) throws OpenViduJavaClientException, OpenViduHttpException, RetryException, InterruptedException;

    boolean isJoined(String conferenceId);

    List<UserProfileDto> getActiveSessions(String conferenceId);

    void changeRemoteConferenceConfig(String conferenceId, ConferenceConfigDto configDto);
}

File: scrumly-ui/src/app/meeting-workspace/services/conference.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from "@angular/common/http";
import { environment } from "../../../enviroments/enviroment";
import { Observable } from "rxjs";
import { ConferenceConfigDto, ConferenceRoomDto } from '../conference/model/conference.model';
import { UserProfileDto } from "../../auth/auth.model";

@Injectable({
  providedIn: 'root'
})
export class ConferenceService {
  private apiUrl = `${ environment.api_url }/conference/api/conference`;

  constructor(private http: HttpClient) {
  }

  joinConference(conferenceId: string): Observable<ConferenceRoomDto> {
    return this.http.post<ConferenceRoomDto>(this.apiUrl + '/' + conferenceId, {});
  }

  isJoined(conferenceId: string): Observable<boolean> {
    return this.http.get<boolean>(this.apiUrl + '/' + conferenceId + '/isJoined', {});
  }

  exitConference(conferenceId: string) {
    return this.http.post<ConferenceRoomDto>(this.apiUrl + '/' + conferenceId + '/exit', {});
  }

  kickFromConference(conferenceId: string, userId: string) {
    return this.http.post<ConferenceRoomDto>(this.apiUrl + '/' + conferenceId + '/' + userId, {});
  }

  closeConference(conferenceId: string) {
    return this.http.post(this.apiUrl + '/' + conferenceId + '/close', {});
  }

  getActiveSessions(conferenceId: string): Observable<UserProfileDto[]> {
    return this.http.get<UserProfileDto[]>(this.apiUrl + '/' + conferenceId + '/sessions', {});
  }

  changeRemoteConferenceConfig(conferenceId: string, config: ConferenceConfigDto) {
    return this.http.put(this.apiUrl + '/' + conferenceId + '/remote/config', config);
  }
}
